### DISCUSSION TITLE
New Dwindle layout dispatchers: `movewindowsplit` and `swapsplit`

### DISCUSSION THREAD

**Discussion by 11xx** [hyprwm/Hyprland#2089] [General]
- `swapsplit`
There's `swapwindow` but it is only directional and for individual windows, it's easier to hit e.g. SUPER + S to swap the two siblings of a node, similar to `togglesplit`.

- `movewindowsplit`
`movewindow` feels limited based on what, on theory, Dwindle can do: On moving a window from and to a split using directions, instead of mostly swaping like `swapwindow` it should split the target window, almost like how it is with a mouse drag&drop.

Here's an example of what is possible on bspwm using [a script](https://gist.github.com/ortango/29f8f4f4d6465c522e82ac60e64bb35d):

https://user-images.githubusercontent.com/89880718/232557718-248ec3c5-d43f-4a3b-a4c5-f8a1b566bc16.mp4

This movement of window in the video is based on:
1. If the direction chosen is within a split/node and there's a sibling in that direction, swap them
2. If the direction chosen is within a split/node and there is no sibling in that direction, either:
 a. Toggle the split orientation/Toggle the split orientation the other way (swapping the split)
 b. Move the window creating a new split on the target node

'2a' wouldn't make much difference and could just be a normal split instead?

I tried going around a script for both the information about the nodes/splits is limited, there's no reliable way of knowing which windows are siblings other than the ones that share a x or y coord, and even so there's no directional split, only the mouse can do something close.

**Comment by vaxerski** [comment]
Yeah, the point no. 2 is planned.

**Comment by truongan** [comment]
Currently the logic for movewindow (https://github.com/hyprwm/Hyprland/blob/50755d26d4662f7392895f35a67502bc8c842a2d/src/managers/KeybindManager.cpp#L1123) and swapwindow (https://github.com/hyprwm/Hyprland/blob/50755d26d4662f7392895f35a67502bc8c842a2d/src/managers/KeybindManager.cpp#L1123) dispatch is very similar when swapping/moving to the a direction on the same workspace. 

I'm curious how far has the plan for  no.2 has progressed 'cause it seem like a very good idea.

**Comment by g-libardi** [comment]
Any news on this? Its a valuable feature that is present in other popular TWM as qtile and i3

**Reply by 11xx** [reply]
This was based on a very early version. They're implemented (`swapsplit` added and `movewindow` behaving as intended). It's pretty good.

Though as a side note I'd still prefer something like bubblemv script for bspwm or what Cosmic Desktop seems to be going for with the directional split and swapping part. But I also didn't feel the need to change this on Hyprland anymore because it's working fine.

**Reply by g-libardi** [reply]
Oh, thank you for your answer, didn't know this was implemented, I will update my config.
